### ISSUE TITLE
"Unnecessary parentheses" on assignment in condition

### ISSUE THREAD

**Issue by 5lava** [kalessil/phpinspectionsea#815]
It's quite common to perform variable assignment in condition. And that's not as bad as it sounds as long as the assignment is explicitly wrapped in parentheses, e.g.:
```php
if (  $x = foo()  ) // bad
if ( ($x = foo()) ) // good
```
First case triggers built-in PHPStorm inspection:

![image](https://user-images.githubusercontent.com/63829/36577270-485dbe16-1824-11e8-9ea3-322091d8c87b.png)

And the second case triggers PHPIEA:

![image](https://user-images.githubusercontent.com/63829/36577282-6ae31260-1824-11e8-9016-d79e01156737.png)

Currently, the only way to break this vicious circle is to extract the assignment outside of the `if` condition (or to disable `UnnecessaryParenthesesInspection`, but that's not why we're here).

It would be great if either
a) the inspection just skipped such cases, or
b) handling of such cases was made configurable.

Thank you!

**Comment by kalessil** [comment]
Apart of that, the case looks like a bug in the native one (but perhaps it intentional), I'm going to decline this request and encourage you to find a solution which satisfies both inspections.

Here some ideas:
- `if (false !== $x = foo()) {}`
- `if (($x = foo()) !== false) {}`
- `if (!empty($x = foo())) {}`
- ` $x = foo(); if ($x) {}`

PS: such cases (cannot satisfy inspections which all making sense) points to code screaming for refactoring/applying clearer intention, hence declining.
